### PR TITLE
Apim 2019 12 01 preview

### PR DESCRIPTION
**Pull request by vfedonkin**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json [hunk 1]
```diff
@@ -3232,21 +3232,26 @@
...
-          "description": "Snippet name.",
-          "readOnly": true
-        },
-        "content": {
-          "type": "string",
-          "description": "Snippet content.",
-          "readOnly": true
-        },
-        "toolTip": {
+        "properties": {
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json]: I am not sure.. why the validation is not failing. But based on the example, this does not have the `properties`.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json [hunk 2]
```diff
@@ -3232,21 +3232,26 @@
...
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/PolicyDescriptionContractProperties",
+          "description": "Policy description contract properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./apimanagement.json#/definitions/Resource"
+        }
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json]: from the example, this does not look like have a `Resource`

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/examples/ApiManagementListApiRevisions.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
       "body": {
         "value": [
           {
+            "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.ApiManagement/service/apimService1/apis/a1;rev=1/revisions/a1;rev=1",
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/examples/ApiManagementListApiRevisions.json]: >": [](start = 15, length = 2)

This is wrong and violates ARM Contract.
Should be like this instead.

{
   `"id"`: "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.ApiManagement/service/apimService1/apis/a1;rev=1/revisions/a1;rev=1",

   `"properties"`: {
            "apiId": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.ApiManagement/service/apimService1/apis/a1;rev=1",
            "apiRevision": "1",
            "createdDateTime": "2018-02-01T22:21:20.467Z",
            "updatedDateTime": "2018-02-01T22:21:20.467Z",
            "isOnline": true,
            "isCurrent": true
     }
}

**Review by solankisamir — CHANGES_REQUESTED** [review]
Can you take a look at the failing validations in the pipeline?

**Comment by vfedonkin** [comment]
failing validations - looks like new version of autorest tool failing (have the same on my dev machine) - previous commit 96132bc wich was successful but now is broken with the same errors when I run validation against it.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/examples/ApiManagementListApiRevisions.json (hunk 3 above)
Comment (vfedonkin) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/examples/ApiManagementListApiRevisions.json]: We have a set of "NonArmStyleEntities" described here:
public static readonly HashSet<string> NonArmStyleEntities = new HashSet<string>(new[] {    ...

Those entities don't have "properties" yet. ARM never complained about this, the only issue was "id property is missing"

**Comment by vfedonkin** [comment]
After upgrading autorest to latest version, validation is fine on my local dev machine.

**Comment by vfedonkin** [comment]
Reverted changes to commit 96132bc to check if validation will be ok for this commit  (was ok last time)

**Review by kshatriyaSam — APPROVED** [review]

**Review by solankisamir — APPROVED** [review]

**Pull request merged**